sentry-android-core/src/main/java/io/sentry/android/core/AndroidOptionsInitializer.java

```diff
@@ -68,9 +73,24 @@ static void init(
     options.addIntegration(new AnrIntegration());
     options.addIntegration(new SessionTrackingIntegration());
 
-    readDefaultOptionValues(options, context);
+    final SharedPreferences sharedPreferences =
+        context.getSharedPreferences(BuildConfig.LIBRARY_PACKAGE_NAME, Context.MODE_PRIVATE);
+
+    String sentryDeviceId;
+    // cache it if not set
+    if (!sharedPreferences.contains(SENTRY_DEVICE_ID)) {
```

Comment:
Is the place to have this code on some package private static class? If so we could just call it from here.

I believe this `contains` call could be replace with the `sharedPreferences.getString(SENTRY_DEVICE_ID, null) != null` to avoid two I/O calls.

Reply:
`SharedPreferences` does in-memory caching sync and it flushes to the disk async., so unless its value has changed, it's not an issue.
I'd separate it in a method or class, yes.